csharp/src/Ice/Ice1NetworkSocketStream.cs

```diff
@@ -113,23 +113,19 @@ internal void ReceivedFrame(Ice1Definitions.FrameType frameType, ArraySegment<by
                 }
 
                 ArraySegment<byte> header;
+                header = buffer[0];
                 if (compressed != null)
                 {
                     buffer = compressed;
-                    header = buffer[0];
```

Comment:
Not related to this PR but we can consider stopping modifying the uncompressed header in `BZip2.Compress` and asign header here to compressed[0]

Reply:
Hmm my fix is actually not correct, header should be assigned after buffer is assigned to compressed... I'll simplify this code.

Reply:
I think your fix works fine because we also modify the uncompressed header, and in the end that is only used to read the compression status...

Reply:
Right but it's still highly confusing! I'll fix Compress to leave untouched the given buffer and setup the compression status in the header only if the message isn't compressed.